packages/shared/client/createReplayClientRecorder.ts

```diff
@@ -37,16 +63,22 @@ export default function createReplayClientRecorder(
 
         const entry: LogEntry = { args: args, isAsync: false, method: prop, result: result };
 
-        logEntries.push(entry);
+        if (findMatch(logEntries, prop, args) === null) {
+          logEntries.push(entry);
+        }
 
         // Unwrap Promise values
         if (result != null && typeof result.then === "function") {
           entry.isAsync = true;
 
+          // flagPendingClientRequest();
```

Comment:
:hand: dead code?

Reply:
I'm not _totally_ convinced we won't need it so for now I've just commented it out.